docs/narr/urldispatch.rst

```diff
@@ -1100,52 +1097,71 @@ was matched (see :class:`pyramid.interfaces.IRoute` for the API of such a route
 object).
 
 ``info['match']`` is useful when predicates need access to the route match.
-For example:
+Imagine you want to define a route when a part of the URL matches some
+specific values.  You could define three view configurations, each one
+with its own ``match_param`` value (see :ref:`predicate_view_args`), but
+you want to think of it as one route, and associate it with one view.
+See this code:
 
 .. code-block:: python
     :linenos:
 
-    def any_of(segment_name, *allowed):
-        def predicate(info, request):
-            if info['match'][segment_name] in allowed:
-                return True
-        return predicate
+    class AnyOfPredicate:
+        def __init__(self, val):
```

Comment:
predicates take two args, `(val, info)` where the second arg is an `IPredicateInfo` object

Reply:
Ah!  I translated the old example, mapping the wrapper function to predicate factory (`__init__`) and the actual predicate function to `__call__`, but I see now that the `ContentTypePredicate` example in `hooks.rst` does have the signature that you mention.
Also the `info` param passed to the predicate factory implements IPredicateInfo, but `info` passed to route predicate instances is a dict with route match details.  It’s clear now!

Reply:
I feel that there should be at least a very short mention of this `info` param in the doc below, even though it’s already explained in the full docs in `hooks.rst`.

Reply:
In tests, predicate classes are defined with ``__init__(self, val, config)``, maybe to avoid the info/info confusion.  I could follow this convention, even though the interface is IPredicateInfo.

Reply:
config is legacy because it used to be the actual configurator, but I don't really care what you call it